### DISCUSSION TITLE
Prevent or show a warning if you overwrite a channel.

### DISCUSSION THREAD

**Discussion by casret** [mclintprojects/actioncable-vue#46] [Ideas]
Because it's pretty easy to overwrite a channel name with a different subscription: https://github.com/mclintprojects/actioncable-vue/blob/4378b80eb1c3ccf211950c80679eacced66de65c/src/cable.js#L71 we've gotten into situations where we either leak subscriptions and get multiple calls to received with the same data, or lost a connection to a channel when trying to share it between components.  It would be good to either warn or raise an exception if you are about to re-use the channel name.  I think an error is probably warranted here since it's not a situation that you can recover from since AFAICT there is no way to now get a reference to the channel that was overwritten.  I'd be happy to write a patch, just wanted to check if I was missing some nuance.

**Comment by mclintprojects** [comment]
Sorry. I didn't see this suggestion but it's a fantastic idea. I'll add this to my backlog.
